pkg/controller/pxc/status.go

```diff
@@ -325,13 +325,12 @@ func (r *ReconcilePerconaXtraDBCluster) appStatus(app api.StatefulApp, podSpec *
 					if !isPXC(app) {
 						status.Ready++
 					} else {
-						isPodWaitingForRecovery, _, err := r.isPodWaitingForRecovery(namespace, pod.Name)
-						if err != nil {
-							return api.AppStatus{}, fmt.Errorf("parse %s pod logs: %v", pod.Name, err)
-						}
-
-						isPodReady := !isPodWaitingForRecovery
+						isPodReady := true
 						if cr170OrGreater {
+							isPodReady, _, err := r.isPodWaitingForRecovery(namespace, pod.Name)
```

Comment:
We need this line
`isPodReady := !isPodWaitingForRecovery`
Because if pod is waiting for recovery, it can't be ready

Reply:
@cap1984  @qJkee 

After the changes:
```
NAME        ENDPOINT                               STATUS         PXC   PROXYSQL   HAPROXY   AGE
some-name   some-name-proxysql.init-deploy-12953   initializing   2     1                    3m38s
some-name   some-name-proxysql.init-deploy-12953   initializing   3     1                    6m38s
some-name   some-name-proxysql.init-deploy-12953   initializing   2     1                    6m53s
some-name   some-name-proxysql.init-deploy-12953   ready          3     1                    7m50s
some-name   some-name-proxysql.init-deploy-12953   initializing   3     1                    8m11s
some-name   some-name-proxysql.init-deploy-12953   initializing   2     1                    8m26s
some-name   some-name-proxysql.init-deploy-12953   initializing   1     1                    8m31s
some-name   some-name-proxysql.init-deploy-12953   initializing         1                    8m46s
some-name   some-name-proxysql.init-deploy-12953   initializing   1     1                    9m38s
some-name   some-name-proxysql.init-deploy-12953   initializing   2     1                    10m
**some-name   some-name-proxysql.init-deploy-12953   ready          3     1                    11m**  we should not have this ready state in case of full cluster crash
some-name   some-name-proxysql.init-deploy-12953   initializing   2     1                    11m
some-name   some-name-proxysql.init-deploy-12953   initializing   1     1                    12m
some-name   some-name-proxysql.init-deploy-12953   initializing   2     1                    12m
some-name   some-name-proxysql.init-deploy-12953   initializing   2     1                    12m
some-name   some-name-proxysql.init-deploy-12953   ready          3     1                    13m
```

before the changes:
```
NAME        ENDPOINT                               STATUS         PXC   PROXYSQL   HAPROXY   AGE
some-name   some-name-proxysql.init-deploy-31271   initializing   1     1                    2m58s
some-name   some-name-proxysql.init-deploy-31271   initializing   2     1                    3m7s
some-name   some-name-proxysql.init-deploy-31271   ready          3     1                    4m13s
some-name   some-name-proxysql.init-deploy-31271   initializing   3     1                    5m
some-name   some-name-proxysql.init-deploy-31271   initializing   1     1                    5m15s
some-name   some-name-proxysql.init-deploy-31271   initializing         1                    5m30s
some-name   some-name-proxysql.init-deploy-31271   initializing   1                          11m
some-name   some-name-proxysql.init-deploy-31271   initializing   1     1                    11m
some-name   some-name-proxysql.init-deploy-31271   initializing   2     1                    12m
some-name   some-name-proxysql.init-deploy-31271   ready          3     1                    12m
```

Reply:
done

Reply:
@qJkee  and it is a good question, why it was not detected by our  test?